Add serde impls to LogLevel and LogLevelFilter

aergonaut opened the pull request:
Fixes #143 

Adds `serde` as an optional dependency and puts the `Serialize` and `Deserialize` traits behind `cfg_attr`. I wasn't sure what version I should specify. 1.0.7 was the latest when I checked, but maybe it's too specific? Would `>= 1.0` be better?

Saw the post on the internals forum, wanted to help out!

alexcrichton commented:
Thanks! Could you also add some CI configuration to test this on Travis/AppVeyor? Also, I think the dependency here is safe to just be `serde = "1.0"` as `>=` is actually a little too eager as it may or may not be compatible with 2.0 (we're not sure if it will be).

dtolnay commented:
@aergonaut are you still working on this or would you need to hand it off to someone else?

aergonaut commented:
I would still like to work on this! I will try to come back to it by the end of the week.

dtolnay commented:
No problem, let us know if you get stuck.

aergonaut commented:
@dtolnay @alexcrichton I think this is ready now! I added `cargo test --features serde` to the Travis and Appveyor configs. I also fixed up the feature flag usage so it actually works.
